code/modules/forensics/forensic_group.dm

```diff
@@ -0,0 +1,192 @@
+
+#define FORENSIC_EVIDENCE_MAX 20
+
+ABSTRACT_TYPE(/datum/forensic_group)
+ABSTRACT_TYPE(/datum/forensic_group/basic_list)
+ABSTRACT_TYPE(/datum/forensic_group/multi_list)
+// Only one of each type of forensics_group should exist per forensic_holder
+// If you want to store multiple groups of the same type, use multiple forensic_holders
+/datum/forensic_group
+	var/category = FORENSIC_GROUP_NONE // An identifier for the group type. Must be unique for each group.
+	var/group_flags = 0 // Flags associated with the whole group. Actual usage may vary by group.
+	var/group_accuracy = 1
+
+	proc/apply_evidence(var/datum/forensic_data/data) // Add a piece of evidence to this group
+		return
+	proc/copy_group(var/datum/forensic_holder/new_holder, var/include_trace = FALSE)
+		return null
+	proc/get_header() // The label that this evidence will be displayed under for scans
+		return SPAN_ALERT("Null Header")
+	proc/remove_evidence(var/datum/forensic_holder/parent, var/removal_flags)
+		return
+	proc/matching_flags(var/flags_A, var/flags_B)
+		return (flags_A & ~FORENSIC_FAKE) == (flags_B & ~FORENSIC_FAKE)
+
+/datum/forensic_group/notes
+	category = FORENSIC_GROUP_NOTE
+	group_flags = FORENSIC_REMOVAL_ALL
+	var/list/datum/forensic_data/basic/notes_list = new/list()
+
+	disposing()
+		src.notes_list.len = 0
+		src.notes_list = null
+		..()
+
+	apply_evidence(var/datum/forensic_data/data)
+		if(istype(data, /datum/forensic_data/basic))
+			var/datum/forensic_data/basic/E = data
```

Comment:
Please don't use single letter variable names unless it's something simple and standard like `turf/T`